### ISSUE TITLE
Bacalhau: rejects linux/amd64 bids despite having linux/amd64 nods

### ISSUE THREAD

**Issue by Nasfame** [bacalhau-project/bacalhau#3513]
```
 bacalhau docker run \
 --input https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf,dst=/inputs/a.pdf \
 \
 docker.io/laciferin/marker:latest\
 -- poetry run python convert_single.py /inputs/a.pdf /outputs/a.md --parallel_factor 1 --max_pages 10 
Job successfully submitted. Job ID: ec30c5db-d47e-4e2c-9a43-b5b3836260af
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

	Communicating with the network  ................  done ✅  0.5s
	   Creating job for submission  ................  err  ❌  1.0s

Error submitting job: not enough nodes to run job. requested: 1, available: 0. 
	Node QmRHRr8c: job rejected
	Node QmbxGSsM: job rejected
	Node QmPLPUUj: job rejected
	Node QmcC3xif: job rejected
	Node QmSD38wH: job rejected
	Node QmeXjeQD: job rejected
Job Results By Node:
• Node QmSD38wH, QmcC3xif, QmbxGSsM, QmPLPUUj, QmeXjeQD, QmRHRr8c: this node does not support the available image platforms [] (supports [linux/amd64]). bid rejected

To get more details about the run, execute:
	bacalhau describe ec30c5db-d47e-4e2c-9a43-b5b3836260af
```

## Debug

```
b docker run alpine -- uname -a
```

```
Linux 50859d58e011 6.2.0-1019-gcp #21~22.04.1-Ubuntu SMP Thu Nov 16 18:18:34 UTC 2023 x86_64 Linux
```
```[tasklist]
### Tasks
- [ ] Debug: https://hub.docker.com/layers/laciferin/marker/latest/images/sha256-172ba19f4998d168860ba84b46d181559dfa1cd09477afe87ebce64d946089fa?context=repo
```

**Comment by rossjones** [comment]
When running this on my M2 Mac, I get the following after applying https://github.com/bacalhau-project/bacalhau/pull/3755 so this may resolve it.

```
Error submitting job: not enough nodes to run job. requested: 1, available: 0. 
        Node node-3: job rejected
        Node node-1: job rejected
        Node node-2: job rejected
Job Results By Node:
• Node node-3, node-2, node-1: rejected bid: this node does not support the available image platforms [linux/amd64] (supports [linux/arm64]). bid rejected
```

**Comment by rossjones** [comment]
@Nasfame I think this may be fixed in main now, if you could retry and close if so, that'd be great.

**Comment by Nasfame** [comment]
Can u confirm that the fix has made it to the latest release?

**Comment by rossjones** [comment]
No, only fixed in main. No worries, can check again when 1.3.1 is released.

**Comment by Nasfame** [comment]
@rossjones getting: 

```
bacalhau docker run \
 --input https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf,dst=/inputs/a.pdf \
 \
 docker.io/laciferin/marker:latest\
 -- poetry run python convert_single.py /inputs/a.pdf /outputs/a.md --parallel_factor 1 --max_pages 10  
Job successfully submitted. Job ID: 7de35486-e46b-4203-9a71-b6c251ebeb77
Checking job status... (Enter Ctrl+C to exit at any time, your job will continue running):

	Communicating with the network  ................  done ✅  0.6s
	   Creating job for submission  ................  done ✅  0.0s
	               Job in progress  ................  err  ❌  31.4s

Error submitting job: preparing arguments: failed to begin download from url https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf: GET https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf giving up after 4 attempt(s)
Job Results By Node:
• Node QmbxGSsM, QmSD38wH, QmPLPUUj, QmeXjeQD, QmRHRr8c, QmeXjeQD, QmSD38wH, QmPLPUUj, QmbxGSsM, QmRHRr8c: preparing arguments: failed to begin download from url https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf: GET https://cdn-careerservices.fas.harvard.edu/wp-content/uploads/sites/161/2023/04/HES_resume-and-letter_2022-final.pdf giving up after 4 attempt(s). execution failed

To get more details about the run, execute:
	bacalhau describe 7de35486-e46b-4203-9a71-b6c251ebeb77

```
Note: the api host is bacalhau production.

**Comment by Nasfame** [comment]
```
 bacalhau docker run \
 --input https://www.forgottenbooks.com/en/download/Short_Stories_Old_and_New_1000313302.pdf,dst=/inputs/a.pdf \
 \
 docker.io/laciferin/marker:latest\
 -- poetry run python convert_single.py /inputs/a.pdf /outputs/a.md --parallel_factor 1
```

```
 bacalhau docker run \
 --input https://shortstorylines.com/wp-content/uploads/2020/01/the-lion-and-the-rabbit-pdf.pdf,dst=/inputs/a.pdf \
 \
 docker.io/laciferin/marker:latest\
 -- poetry run python convert_single.py /inputs/a.pdf /outputs/a.md --parallel_factor 1 --max_pages=10 
```

**Comment by Nasfame** [comment]
Resolved and working in release - `v1.3.0`
